### ISSUE TITLE
Large Text Entry

### ISSUE THREAD

**Issue by 0x414a** [fyne-io/fyne#5059]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

![QQ_1723130286687](https://github.com/user-attachments/assets/d09bceda-8d5d-4af3-8b0e-20ce4febce1d)

https://github.com/user-attachments/assets/53c4c302-ecb5-4c75-ac03-5d6f157da58c

### Is it possible to construct a solution with the existing API?

Some applications require a text editing box for extremely long content, but the current Entry widget is very sluggish when rendering large amounts of text. I have customized a canvas widget to support extensive text input, which dynamically adjusts its height based on content changes using container.NewVScroll. However, I haven't fully implemented cursor control and basic input functionalities as efficiently as the standard Entry. Due to time constraints, I haven't been able to refine these features. I see that Fyne has progressed to version 2.5.0 now. My approach involves creating a virtual container to handle large text portions and rendering them based on scroll positions. While my CustomCanvasEntryText offers basic editing capabilities, it doesn't match the completeness of Entry. I've found developing this from scratch quite time-consuming. I'm uncertain if Fyne's current or future versions will support extended content in text inputs.

### Describe the solution you'd like to see.

Certainly, if Fyne currently has no plans to support input boxes for extremely long content, I wonder if the Entry widget could expose a scroll interface. This would allow experimenting with virtualization directly on the Entry to achieve partial rendering of extensive text, thus addressing the performance and resource consumption issues associated with rendering large text. By exposing the scroll interface of Entry, I believe it should be possible to customize and implement my own rendering using canvas (leveraging the Entry directly, avoiding the need to separately draw an input box).

![QQ_1723131390628](https://github.com/user-attachments/assets/c7d61fe8-e65d-4a1e-83e6-4733a269323a)

**Comment by 0x414a** [comment]
Translated by GPT-4, the description may not be precise enough.

**Comment by andydotxyz** [comment]
Improving the performance of Entry widget should be done in the widget itself not by exposing more APIs so that apps can solve the problem manually.

The intention is that it would use the same cache / re-use semantics as the list widget internally.

**Comment by 0x414a** [comment]
It seems I didn't express myself clearly enough. What I'm aiming to achieve is a large text input box, not simply adding items to a list. Indeed, in the video, I forgot to demonstrate features like deleting and entering new lines, similar to what an `Entry` widget would do. The issue with `Entry` is that when there is too much content, any modification to any part of the content triggers a full re-render of the entire `Entry`, and even the initial loading of a large amount of text can be very laggy. In my video, I customized a text input box using a `Canvas`, implementing a scrolling rendering mechanism similar to tree-based scrolling, but my implementation wasn't perfect. So, I was wondering if there will be future improvements to address the rendering issues with large amounts of text in `Entry`, or if we could customize it directly on top of the `Entry` foundation.



---Original---
From: "Andy ***@***.***&gt;
Date: Fri, Aug 9, 2024 04:54 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [fyne-io/fyne] Large Text Entry (Issue #5059)




 
Improving the performance of Entry widget should be done in the widget itself not by exposing more APIs so that apps can solve the problem manually.
 
The intention is that it would use the same cache / re-use semantics as the list widget internally.

**Comment by andydotxyz** [comment]
> The issue with `Entry` is that when there is too much content, any modification to any part of the content triggers a full re-render of the entire `Entry`, and even the initial loading of a large amount of text can be very laggy

Yes indeed. And we should resolve that in Entry instead of adding new APIs etc.

> So, I was wondering if there will be future improvements to address the rendering issues with large amounts of text in `Entry`

Yes

**Comment by 0x414a** [comment]
Looking forward to official Fyne support for extra-long content in input fields.



---Original---
From: "Andy ***@***.***&gt;
Date: Fri, Aug 9, 2024 15:43 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [fyne-io/fyne] Large Text Entry (Issue #5059)




  
The issue with Entry is that when there is too much content, any modification to any part of the content triggers a full re-render of the entire Entry, and even the initial loading of a large amount of text can be very laggy
  
Yes indeed. And we should resolve that in Entry instead of adding new APIs etc.
  
So, I was wondering if there will be future improvements to address the rendering issues with large amounts of text in Entry
  
Yes
